src/orion/core/worker/multi_task_algo.py

```diff
@@ -0,0 +1,551 @@
+            n_incompatible_trials = 0
+
+            for trial in trials:
+                try:
+                    # TODO: Use something smarter than `trial_to_tuple`, since it
+                    # doesn't adjust the points if it's just missing a default value.
+                    point = trial_to_tuple(trial=trial, space=self.space)
+                    # Drop the point if it doesn't fit inside the current space.
+                    # TODO: Do we want to 'translate' the point in this case?
+                    # if self.translate_points:
+                    #     point = self.translator.translate(point)
+                    if point not in self.space:
+                        raise ValueError
+                    # Add the task id to the point:
+                    # The task ID of the given experiment is `i + 1`, so that the task
+                    # id 0 is reserved for the trials of the current task.
+                    # TODO: This assumes that the trials from the knowledge base don't
+                    # include those of the current experiment.
+                    task_id = i + 1
+
+                    # point = self._add_task_id(point, task_id)
+                    # trial = self._add_task_id(trial, task_id)
+
+                    compatible_trials.append(trial)
+                    compatible_points.append(point)
+                    task_ids.append(task_id)
+
+                except ValueError as e:
+                    log.error(f"Can't reuse trial {trial}: {e}")
+                    n_incompatible_trials += 1
+
+            total_trials = len(trials)
+            n_compatible_trials = total_trials - n_incompatible_trials
+            log.info(
+                f"Experiment {experiment_info.name} has {len(trials)} trials in "
+                f"total, out of which {n_compatible_trials} were found to be "
+                f"compatible with the target experiment."
+            )
+
+        if not compatible_trials:
+            log.info("No compatible trials detected.")
+            return
+
+        # Only keep trials that are new.
+        # TODO: Check that the task dim doesnt interfere with this:
+        new_trials, new_points, new_task_ids = list(
+            zip(
+                *[
+                    (trial, point, task_id)
+                    for trial, point, task_id in zip(
+                        compatible_trials, compatible_points, task_ids
+                    )
+                    if infer_trial_id(point) not in self.unwrapped._warm_start_trials
+                ]
+            )
+        )
+        if not new_trials:
+            log.info("No new new warm-starting trials detected.")
+            return
+
+        with self.algorithm.warm_start_mode():
+            new_points_with_task_ids = [
+                self._add_task_id(point, task_id)
+                for point, task_id in zip(new_points, new_task_ids)
+            ]
+            new_trials_with_task_ids = [
```

Comment:
Before release v0.2.0 we will drop points and use trials instead in observe/suggest, so here we could only use trials and forget about points.